Classes/UpgradeWizard/FieldLogoMigrationWizard.php

```diff
@@ -0,0 +1,151 @@
+     */
+    protected function getRecordsForMigration(bool $onlyCount = true): int|array
+    {
+        $connectionPool = $this->getConnectionPool();
+        $queryBuilder = $connectionPool->getQueryBuilderForTable('sys_file_reference');
+        $queryBuilder->getRestrictions()->removeAll();
+
+        try {
+            // Build the base query
+            $queryBuilder->from('sys_file_reference')
+                ->orWhere(
+                    $queryBuilder->expr()->eq(
+                        'tablenames',
+                        $queryBuilder->createNamedParameter('tx_daycarecenters_domain_model_kita', \PDO::PARAM_STR)
+                    ),
+                    $queryBuilder->expr()->eq(
+                        'tablenames',
+                        $queryBuilder->createNamedParameter('tx_daycarecenters_domain_model_holder', \PDO::PARAM_STR)
+                    )
+                )
+                ->andWhere(
+                    $queryBuilder->expr()->eq(
+                        'fieldname',
+                        $queryBuilder->createNamedParameter('logo', \PDO::PARAM_STR)
+                    )
+                );
+
+            // Modify the query based on the $onlyCount parameter
+            if ($onlyCount) {
+                // Select the count of matching records
+                $queryBuilder->selectLiteral('COUNT(uid) as count');
+                $result = $queryBuilder->executeQuery();
+
+                return (int)$result->fetchOne();
+            }
+
+            // Select the required fields for matching records
+            $queryBuilder->select('uid', 'fieldname', 'tablenames');
+            $result = $queryBuilder->executeQuery();
+
+            return $result->fetchAllAssociative();
+        } catch (DBALException $e) {
+            throw new \RuntimeException(
+                'Database query failed. Error was: ' . $e->getMessage(),
+                1596705829853
+            );
+        }
+    }
+
+    private function migrateRecords(): void
+    {
+        $queryBuilder = $this->getConnectionPool()->getQueryBuilderForTable('sys_file_reference');
+        $affectedRows = $queryBuilder
+            ->update('sys_file_reference')
+            ->orWhere(
+                $queryBuilder->expr()->eq(
+                    'tablenames',
+                    $queryBuilder->createNamedParameter('tx_daycarecenters_domain_model_kita', \PDO::PARAM_STR)
+                ),
+                $queryBuilder->expr()->eq(
+                    'tablenames',
+                    $queryBuilder->createNamedParameter('tx_daycarecenters_domain_model_holder', \PDO::PARAM_STR)
+                )
+            )
+            ->andWhere(
+                $queryBuilder->expr()->eq(
+                    'fieldname',
+                    $queryBuilder->createNamedParameter('logo', \PDO::PARAM_STR)
+                )
+            )
+            ->set('fieldname', 'logos')
+            ->executeStatement();
```

Comment:
`orWhere` is so nasty, it contains another wrapping of where clauses and has curious handling about previous added where clauses. Don't use it.

```suggestion
        $affectedRows = $queryBuilder
            ->update('sys_file_reference')
            ->set('fieldname', 'logos')
            ->where(
                $queryBuilder->expr()->eq(
                    'fieldname',
                    $queryBuilder->createNamedParameter('logo', \PDO::PARAM_STR)
                ),
                $queryBuilder->expr()->or(
                    $queryBuilder->expr()->eq(
                        'tablenames',
                        $queryBuilder->createNamedParameter('tx_daycarecenters_domain_model_kita', \PDO::PARAM_STR)
                    ),
                    $queryBuilder->expr()->eq(
                        'tablenames',
                        $queryBuilder->createNamedParameter('tx_daycarecenters_domain_model_holder', \PDO::PARAM_STR)
                    )
                )
            )
            ->executeStatement();
```